packages/wonder-blocks-switch/src/components/switch.tsx

```diff
@@ -0,0 +1,224 @@
+import * as React from "react";
+import {StyleSheet} from "aphrodite";
+
+import Clickable, {
+    ClickableRole,
+    ClickableState,
+} from "@khanacademy/wonder-blocks-clickable";
+import Color, {mix} from "@khanacademy/wonder-blocks-color";
+import {
+    UniqueIDProvider,
+    View,
+    addStyle,
+} from "@khanacademy/wonder-blocks-core";
+
+type Props = {
+    /**
+     * A label for the switch.
+     * If a visible label is not provided, aria-label should be used to
+     * describe the purpose of the switch.
+     */
+    ariaLabel?: string;
+    /**
+     * Whether this compoonent is checked.
+     */
+    checked: boolean;
+    /**
+     * Whether the switch is disabled. Defaults to `false`.
+     */
+    disabled?: boolean;
+    /**
+     * The unique identifier for the switch.
+     */
+    id?: string;
+    /**
+     * Function to call when the switch is clicked.
+     * @param newCheckedState
+     * @returns {unknown}
+     */
+    onChange?: (newCheckedState: boolean) => unknown;
+    /**
+     * Test ID used for e2e testing.
+     */
+    testId?: string;
+};
+
+const StyledSpan = addStyle("span");
+
+const Switch = React.forwardRef(function Switch(
+    props: Props,
+    ref: React.ForwardedRef<HTMLButtonElement>,
+) {
+    const {ariaLabel, checked, disabled = false, id, onChange, testId} = props;
+
+    const renderSwitch = (
+        uniqueId?: string,
+        eventState?: ClickableState,
+        sharedProps?: any,
+    ) => {
+        return (
+            <View
+                id={uniqueId}
+                style={[
+                    sharedStyles.switch,
+                    stateStyles.switch,
+                    !disabled &&
+                        (eventState?.hovered || eventState?.focused) &&
+                        sharedStyles.outline,
+                    disabled &&
+                        eventState?.focused &&
+                        sharedStyles.disabledOutline,
+                    !disabled && eventState?.pressed && stateStyles.pressed,
+                ]}
+                {...sharedProps}
+            >
```

Comment:
I think you might need to add an `aria-checked` here. Currently if you try to use a screen reader in your storybook examples, it doesn't read out the `checked` value of the switch.

Reply:
hm... it should already be in the shared props. i was able to get it to read it out?
<img width="684" alt="image" src="https://github.com/Khan/wonder-blocks/assets/77523470/91425481-34e1-4c9c-bb0d-8133b2ef6890">

Reply:
Oh hm... I was trying it on the `Inside Cell` story. I was hoping it would work kind of like the [WithLabel](https://khan.github.io/wonder-blocks/?path=/story/form-checkbox--with-label) story on checkbox. This might be something that would need to be updated in `Cell` though, which would be out of scope for this PR.

Reply:
ooh. yea that would be something that needs to be specified on `Cell`. i'm not sure if we can enforce it from the `Switch` side of things but i'll add a note in the story about it.